nodes/persistence-node/src/getIpfsFileContents.ts

```diff
@@ -1,18 +1,34 @@
 import * as IPFS from 'ipfs-core';
+import { NotFoundError } from './types/NotFoundError';
 
 export const getIpfsFileContents = async (ipfs: IPFS.IPFS, hash: string): Promise<Buffer> => {
-  const stream = ipfs.cat(hash);
+  try {
+    const stream = ipfs.cat(hash);
 
-  let data: Uint8Array = new Uint8Array();
+    let data: Uint8Array = new Uint8Array();
 
-  for await (const chunk of stream) {
-    const temp = new Uint8Array(data.length + chunk.length);
-    temp.set(data);
-    temp.set(chunk, data.length);
-    data = temp;
-  }
+    for await (const chunk of stream) {
+      const temp = new Uint8Array(data.length + chunk.length);
+      temp.set(data);
+      temp.set(chunk, data.length);
+      data = temp;
+    }
+
+    const buffer = Buffer.from(data);
 
-  const buffer = Buffer.from(data);
+    return buffer;
+  } catch (err: any) {
+    const errorMessage: string | null = typeof err?.message === "string" ? err.message : null;
+    
+    if (errorMessage?.includes("Invalid CID version")) {
```

Comment:
If we really have to determine errors from string messages, I would cover it with some unit tests, so we at least have some way of noticing if those error messages change in the future.

Reply:
Good point. I implemented tests to cover 404 cases.